build.sbt

```diff
@@ -135,31 +54,23 @@ lazy val core = crossProject(JSPlatform, JVMPlatform)
       if (tlIsScala3.value) Nil else Seq("org.scala-lang" % "scala-reflect" % scalaVersion.value)
     },
     scalacOptions ++= {
-      val isScala211 = CrossVersion.partialVersion(scalaVersion.value).contains((2, 11))
       // this code seems to trigger a bug in 2.11 pattern analysis
-      if (isScala211) List("-Xno-patmat-analysis") else Nil
+      if (isScala211.value) List("-Xno-patmat-analysis") else Nil
     },
     mimaPreviousArtifacts := {
-      val isScala211 = CrossVersion.partialVersion(scalaVersion.value).contains((2, 11))
-      if (isScala211) Set.empty else mimaPreviousArtifacts.value
+      if (isScala211.value) Set.empty else mimaPreviousArtifacts.value
     },
     mimaBinaryIssueFilters ++= {
       /*
        * It is okay to filter anything in Impl or RadixNode which are private
        */
       if (tlIsScala3.value)
         List(
-          ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.RadixNode.children"),
-          ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.RadixNode.fsts"),
-          ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.RadixNode.prefixes"),
-          ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.RadixNode.word"),
-          ProblemFilters.exclude[FinalClassProblem]("cats.parse.RadixNode"),
-          ProblemFilters.exclude[IncompatibleMethTypeProblem]("cats.parse.Parser#State.error_="),
-          ProblemFilters.exclude[IncompatibleMethTypeProblem]("cats.parse.RadixNode.this"),
-          ProblemFilters.exclude[IncompatibleResultTypeProblem]("cats.parse.Parser#State.error"),
           ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.Parser#Error.fromProduct"),
           ProblemFilters.exclude[IncompatibleResultTypeProblem]("cats.parse.Parser#Error.unapply"),
-          ProblemFilters.exclude[MissingTypesProblem]("cats.parse.Parser$Error$")
+          ProblemFilters.exclude[MissingTypesProblem]("cats.parse.Parser$Error$"),
+          ProblemFilters.exclude[IncompatibleResultTypeProblem]("cats.parse.Parser#Error.unapply"),
+          ProblemFilters.exclude[DirectMissingMethodProblem]("cats.parse.Parser#Error.fromProduct")
```

Comment:
MiMa had some improvements for Scala 3 so we can remove some unnecessary filters.